docs/10_execution_architecture.md

```diff
@@ -0,0 +1,296 @@
+# Execution Architecture
+
+## Overview
+
+This document describes the architecture of the execution module (`packages/mobile-native-mcp-server/src/execution/`), which provides an abstraction for executing long running node commands with real-time progress reporting in action nodes. The design emphasizes encapsulation of platform-specific logic, clear separation of concerns, and dependency injection for testability.
+
+The execution module is organized into two layers:
+- **Generic Execution** (`src/execution/`): Reusable components for command execution and progress reporting
+- **Build-Specific Execution** (`src/execution/build/`): Build-specific command creation and execution logic
```

Comment:
note I reference "build" throughout this doc but I'm intentionally vague around the details. While the build node is the motivating feature I wanted to keep the emphasis on the generic problem of nodes executing long running commands.

the actual "build" command will be scoped to another PR as there's going to be a lot of subtleties there that warrant their own review for extra scrutiny